endpoint.cpp

```diff
@@ -236,8 +238,11 @@ int Endpoint::read_msg(struct buffer *pbuf, int *target_sysid)
     if (msg_entry == nullptr) {
         log_debug("No message entry for %u", msg_id);
     } else {
-        if (msg_entry->flags & MAV_MSG_ENTRY_FLAG_HAVE_TARGET_SYSTEM)
-            *target_sysid = payload[msg_entry->target_system_ofs];
+        if (msg_entry->flags & MAV_MSG_ENTRY_FLAG_HAVE_TARGET_SYSTEM) {
+            // if target_system is 0, it may have been trimmed out on mavlink2
```

Comment:
so the message actually has a target_system. Shouldn't you set *target_sysid to 0 in this case? Then it's clear -1 means "no target sysid" and 0 means broadcast.